packages/spotlight/Spottable/Spottable.js

```diff
@@ -176,58 +174,26 @@ const Spottable = hoc(defaultConfig, (config, Wrapped) => {
 			super(props);
 			this.isFocused = false;
 			this.isHovered = false;
+
+			this.state = {
+				spottableDisabled: false
+			};
 		}
 
-		componentDidMount () {
-			// eslint-disable-next-line react/no-find-dom-node
-			this.node = ReactDOM.findDOMNode(this);
+		componentWillReceiveProps () {
+			this.setState((state, nextProps) => {
+				const spottableDisabled = this.isFocused && nextProps.disabled || nextProps.spotlightDisabled;
+				return !this.state.spottableDisabled && spottableDisabled ? {spottableDisabled} : null;
+			});
 		}
 
-		componentDidUpdate (prevProps) {
-			// if the component is focused and became disabled,
-			if (this.isFocused && (
-				(!prevProps.disabled && this.props.disabled) ||
-				(!prevProps.spotlightDisabled && this.props.spotlightDisabled)
-			)) {
-				forward('onSpotlightDisappear', null, this.props);
+		componentDidUpdate (_, prevState) {
+			// if the component is focused and became disabled
+			if (!prevState.spottableDisabled && this.state.spottableDisabled) {
 				if (lastSelectTarget === this) {
 					selectCancelled = true;
 					forward('onMouseUp', null, this.props);
 				}
-
-				// if spotlight didn't move, find something else to spot starting from here
-				const current = Spotlight.getCurrent();
-				if (!current || this.node === current) {
-					this.node.blur();
-
-					if (!Spotlight.getPointerMode()) {
-						getContainersForNode(this.node).reverse().reduce((found, id) => {
-							return found || Spotlight.focus(id);
-						}, false);
-					}
-				}
-			}
-
-			// if the component became enabled, notify spotlight to enable restoring "lost" focus
```

Comment:
I don't think we can remove this block of code which is handling the inverse (changing from disabled to enabled) of the change proposed. Doing so would likely cause regressions of #1254 and #1023.